From 27fb4b4c6ea6eeae39e1a202e0c4455fe9865a85 Mon Sep 17 00:00:00 2001
From: Justin Suess <utilityemal77@gmail.com>
Date: Mon, 29 Jun 2026 17:38:57 -0400
Subject: [PATCH 10/13] arm64: dts: allwinner: x96q-pro-plus: enable gmac1
 ethernet

Enable the RGMII ethernet on the X96Q Pro+ via gmac1: the external PHY at
MDIO address 1 (with its reset GPIO and timings), rgmii-id mode with
MAC-side tx/rx internal delays, the cldo4 PHY supply (with a 150ms
enable ramp delay) and the internal 25 MHz reference clock path. Also
add the ethernet0 alias.
---
 .../dts/allwinner/sun55i-h728-x96qpro+.dts    | 26 +++++++++++++++++++
 1 file changed, 26 insertions(+)

diff --git a/arch/arm64/boot/dts/allwinner/sun55i-h728-x96qpro+.dts b/arch/arm64/boot/dts/allwinner/sun55i-h728-x96qpro+.dts
index a96927fbdadd..56cd470cbc91 100644
--- a/arch/arm64/boot/dts/allwinner/sun55i-h728-x96qpro+.dts
+++ b/arch/arm64/boot/dts/allwinner/sun55i-h728-x96qpro+.dts
@@ -12,6 +12,7 @@ / {
 	compatible = "amediatech,x96q-pro-plus", "allwinner,sun55i-h728";
 
 	aliases {
+		ethernet0 = &gmac1;
 		serial0 = &uart0;
 	};
 
@@ -54,11 +55,35 @@ &ehci1 {
 	status = "okay";
 };
 
+&gmac1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&rgmii1_pins>, <&rgmii1_25m_pin>;
+	phy-mode = "rgmii-id";
+	phy-handle = <&ext_rgmii_phy>;
+	phy-supply = <&reg_cldo4>;
+
+	tx-internal-delay-ps = <200>;
+	rx-internal-delay-ps = <500>;
+	allwinner,use-25m-clk;
+
+	status = "okay";
+};
+
 &gpu {
 	mali-supply = <&reg_dcdc2>;
 	status = "okay";
 };
 
+&mdio1 {
+	ext_rgmii_phy: ethernet-phy@1 {
+		compatible = "ethernet-phy-ieee802.3-c22";
+		reg = <1>;
+		reset-gpios = <&pio 9 16 GPIO_ACTIVE_LOW>;
+		reset-assert-us = <10000>;
+		reset-deassert-us = <150000>;
+	};
+};
+
 &mmc0 {
 	vmmc-supply = <&reg_vcc3v3>;
 	cd-gpios = <&pio 5 6 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>; /* PF6 */
@@ -212,6 +237,7 @@ reg_cldo4: cldo4 {
 				regulator-min-microvolt = <3300000>;
 				regulator-max-microvolt = <3300000>;
 				regulator-name = "vcc-eth-phy";
+				regulator-enable-ramp-delay = <150000>;
 			};
 
 			reg_cpusldo: cpusldo {
-- 
2.54.0

